internal/server/providers/oidc.go

```diff
@@ -43,6 +42,7 @@ type OIDCClient interface {
 	ExchangeAuthCodeForProviderTokens(ctx context.Context, code string) (*IdentityProviderAuth, error)
 	RefreshAccessToken(ctx context.Context, providerUser *models.ProviderUser) (accessToken string, expiry *time.Time, err error)
 	GetUserInfo(ctx context.Context, providerUser *models.ProviderUser) (*UserInfoClaims, error)
+	Provider() *models.Provider // returns the provider this client was created for
```

Comment:
I could be wrong, but it seems like we only ever use the `ProviderID` from this return value. The one place we pass the full model (`AssignIdentityToGroups`) ends up only using the ID.

Could we make this `ProviderID() uid.ID` and change `AssignIdentityToGroups` to accept just the ID? Or do you anticipate needing more than the ID in the future?

Reply:
You're right, even further we didnt need the provider ID on the client at all after making some of your suggested changes. It was a workaround for some spots where I needed to lookup the provider that dont exist now after more refactoring. Removed this.